client/components/tinymce/index.jsx

```diff
@@ -416,9 +416,11 @@ module.exports = React.createClass( {
 		if ( this._editor ) {
 			const { mode } = this.props;
 			this._editor.setContent( wpautop( content ), { ...args, mode } );
-
-			// clear the undo stack to ensure that we don't have any leftovers
-			this._editor.undoManager.clear();
+			const resetHistory = args && args.initial;
```

Comment:
I personally like default values approach. 

If we merge with `mode` to a variable. It should be another variable I think to avoid confusion with the param. 

Let me update this